packages/orbit-components/src/ErrorFormTooltip/Tooltip/index.js

```diff
@@ -131,40 +157,61 @@ StyledCloseButton.defaultProps = {
 
 const ErrorFormTooltip = ({
   onShown,
-  labelRef,
   dataTest,
   inputSize,
-  iconRef,
   children,
   shown,
-  isHelp = false,
+  referenceElement,
   inlineLabel,
+  isHelp = false,
   id,
 }: Props): React.Node => {
   const contentRef = React.useRef(null);
-  const tooltipRef = React.useRef<HTMLElement | null>(null);
-
-  const dimensions = useDimensions({
-    labelRef,
-    iconRef,
-    contentRef: tooltipRef,
-    children,
-    inlineLabel,
+  const { rtl } = useTheme();
+  const [tooltipRef, setTooltipRef] = React.useState(null);
+  const [arrowRef, setArrowRef] = React.useState(null);
+
+  const resolveOffset = () => {
+    if (inlineLabel) {
+      if (inputSize === "small") return [rtl ? 10 : -14, 7];
+      return [rtl ? 6 : -6, 6];
+    }
+    return [rtl ? 10 : -10, 7];
+  };
+
+  const { styles, attributes: attrs, update } = usePopper(referenceElement?.current, tooltipRef, {
+    placement: rtl ? "top-end" : "top-start",
+    modifiers: [
+      {
+        name: "offset",
+        options: {
+          offset: resolveOffset(),
```

Comment:
It would probably be a good idea to call `update` when the result of this function changes, in that case you'd have to memoize `resolveOffset`:

```js
const resolveOffset = React.useCallback(() => {
  if (inlineLabel) {
    if (inputSize === "small") return [rtl ? 10 : -14, 7];
    return [rtl ? 6 : -6, 6];
  }
  return [rtl ? 10 : -10, 7];
}, [inlineLabel, inputSize, rtl]);

// `usePopper()` call

React.useEffect(() => {
  if (update) update();
}, [update, resolveOffset);
```